dl/src/main/scala/com/intel/analytics/bigdl/optim/Optimizer.scala

```diff
@@ -61,8 +62,15 @@ abstract class Optimizer[T: ClassTag, D](
   }
 
   def setCheckpoint(path: String, trigger: Trigger): this.type = {
-    require(Files.isDirectory(Paths.get(path)), s"$path is not a folder")
-    this.checkpointPath = Some(path)
+    if (path.startsWith("/") || path.startsWith("file:")) {
```

Comment:
We need to support three kinds of path:
For the path `apath` --> [hdfs://]user/username/apath
For the path `/apath` --> [hdfs://]apath
For the path `file:///apath` --> 'file:///apath'